client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-global-styles-upgrade-modal.ts

```diff
@@ -0,0 +1,57 @@
+import { useState } from 'react';
+import { urlToSlug } from 'calypso/lib/url';
+import { useSite } from '../../../../../hooks/use-site';
+import { useSiteSlugParam } from '../../../../../hooks/use-site-slug-param';
+
+interface Props {
+	recordTracksEvent: ( eventName: string, eventProps?: { [ key: string ]: unknown } ) => void;
+	onSubmit: () => void;
+}
+
+const useGlobalStylesUpgradeModal = ( { recordTracksEvent, onSubmit }: Props ) => {
+	const [ isOpen, setIsOpen ] = useState( false );
+	const site = useSite();
+	const siteSlug = useSiteSlugParam();
+	const siteUrl = siteSlug || urlToSlug( site?.URL || '' ) || '';
+
+	const openModal = () => {
+		recordTracksEvent( 'calypso_signup_pattern_assembler_global_styles_gating_modal_show' );
+		setIsOpen( true );
+	};
+
+	const closeModal = () => {
+		recordTracksEvent(
+			'calypso_signup_pattern_assembler_global_styles_gating_modal_close_button_click'
+		);
+		setIsOpen( false );
+	};
+
+	const checkout = () => {
+		recordTracksEvent(
+			'calypso_signup_pattern_assembler_global_styles_gating_modal_checkout_button_click'
+		);
+
+		// When the user is done with checkout, send them back to the current url
+		const destUrl = new URL( window.location.href );
```

Comment:
It has chance that the selected design is reset if the user stays on the checkout page for a long time 🤔